Ballot SC-070: Clarify the use of DTPs for domain control validation

aarongable opened the pull request:
Add a paragraph to BRs Section 3.2.2 clarifying that all Domain Control Validation DNS queries must be conducted by the CA itself, without the use of third-party recursive resolvers. Similarly clarify that looking up information for Domain Contacts and IP Address Contacts must also be done without third-party services.

Require that CAA checks be performed by the CA itself, and not delegated to a third party.

Fixes https://github.com/cabforum/servercert/issues/274

XolphinMartijn commented on docs/BR.md:
```diff
@@ -704,6 +704,8 @@ The CA SHALL confirm that prior to issuance, the CA has validated each Fully-Qua
 1. When the FQDN is not an Onion Domain Name, the CA SHALL validate the FQDN using at least one of the methods listed below; and
 2. When the FQDN is an Onion Domain Name, the CA SHALL validate the FQDN in accordance with Appendix B.
 
+All DNS queries conducted in the course of validation MUST be made from the CA to authoritative nameservers, i.e. without the use of recursive resolvers operated outside the CA's audit scope.
```
Could we move this to 3.2.2 instead, and rephrase as:

"All DNS queries conducted in the course of validation in Section 3.2.2.4, Section 3.2.2.5 and Section 3.2.2.8 MUST be made from the CA to authoritative nameservers, i.e. without the use of recursive resolvers operated outside the CA's audit scope."

MadZeg replied:
Today, a CA may use an external DNS resolver for the purpose of checking CAA according to 3.2.2.8 as long as this is done by a DTP (and according to the requirements for using a DTP) . Do we really want to change this? I think this is out-of-scope for this ballot. See BRF 1.3.2 for details - we should restrict the ballot to 3.2.2.4 and 3.2.2.5.

MadZeg replied:
I understand and agree, then it makes sense to include CAA lookups in this ballot.

aarongable reviewed (commented):
Feedback from Validation subcommittee meeting 2024-01-25:
- "All DNS queries conducted in the course of..." sounds like it could include, say, DNS queries made by a validation specialist's browser while looking up the requirements they have to abide by. One suggestion is "All DNS queries _required to be_ conducted in the course of...", but this then perhaps exempts DNS SOA lookups to get Domain Contact information because those are technically optional (could be replaced by WHOIS lookups).
- The phrasing of "from the allowed sources" doesn't make it immediately clear that those allowed sources are listed in the Section 1.6 definition of Domain Contact.
- The sentences added to 3.2.2.4 and 3.2.2.5 don't cover CAA lookups, which are described in 3.2.2.8. We need to ensure that CAA is covered, and probably end up resolving https://github.com/cabforum/servercert/issues/274 at the same time.

aarongable commented:
> * "All DNS queries conducted in the course of..." sounds like it could include, say, DNS queries made by a validation specialist's browser while looking up the requirements they have to abide by. One suggestion is "All DNS queries _required to be_ conducted in the course of...", but this then perhaps exempts DNS SOA lookups to get Domain Contact information because those are technically optional (could be replaced by WHOIS lookups).

Addressed in the latest commit by saying "in the course of satisfying the requirements of 3.2.2.4, 3.2.2.5, and 3.2.2.8".

> * The phrasing of "from the allowed sources" doesn't make it immediately clear that those allowed sources are listed in the Section 1.6 definition of Domain Contact.

Addressed in the latest commit by modifying the definition of Domain Contact to say what one _is_ rather than where the info comes from, and explicitly laying out the requirements for how to obtain a Domain Contact's info in Section 3.2.2.

> * The sentences added to 3.2.2.4 and 3.2.2.5 don't cover CAA lookups, which are described in 3.2.2.8. We need to ensure that CAA is covered, and probably end up resolving [BRs: Clarify whether 3.2.2.8 can be delegated #274](https://github.com/cabforum/servercert/issues/274) at the same time.

Addressed in the latest commit by moving the verbiage to Section 3.2.2, and explicitly calling out 3.2.2.8 as one of the covered sections.

dzacharo commented on docs/BR.md:
```diff
@@ -199,7 +199,7 @@ Certification Authority (CA) is defined in [Section 1.6](#16-definitions-and-acr
 
 ### 1.3.2 Registration Authorities
 
-With the exception of [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control) and [Section 3.2.2.5](#3225-authentication-for-an-ip-address), the CA MAY delegate the performance of all, or any part, of [Section 3.2](#32-initial-identity-validation) requirements to a Delegated Third Party, provided that the process as a whole fulfills all of the requirements of [Section 3.2](#32-init ...
```
Although this seems to have been implicitly required, if we are to make it explicit, we probably will need some sort of effective date (something like ~30 days after the IPR period clears).

aarongable replied:
I've added an effective date of 2024-05-15, expected to be between 1 and 2 months after the IPR period and nicely aligned with the "middle of odd months" idea.

dzacharo reviewed (commented):
Changes look good to me. Please check the comment about an effective date for 3.2.2.8.
